src/context/SecretsContext.tsx

```diff
@@ -40,29 +42,45 @@ export const SecretsProvider: React.FC<SecretsProviderProps> = ({
   children,
 }) => {
   const [secrets, setSecrets] = useState<Secret[]>([])
+  const { user } = useAuth()
 
   useEffect(() => {
+    let mounted = true
```

Comment:
what's this?

Reply:
I was getting an error "Can’t perform a React state update on an unmounted component." which I guess was happening because of state changes while the component was not yet mounted.
I have applied the fix as mentioned [here](https://www.debuggr.io/react-update-unmounted-component/)

Reply:
this will not fix that issue. can you see why?

Reply:
Not sure why it won't work.
My understanding was that we are updating the state if the `mounted` variable is `true`, and if the component is un-mounted the `mounted` variable will be set to false and it won't enter the `if` block.